PFOffer.sol

```diff
@@ -98,6 +118,7 @@ contract PFOffer {
         oneTimeCosts = _oneTimeCosts;
         minDailyWithdrawLimit = _minDailyWithdrawLimit;
         dailyWithdrawLimit = _minDailyWithdrawLimit;
+        proposalID = _proposalID;
```

Comment:
The contractors doesn't know the proposal id at this point at it might take a while until it's whitelisted. They should have a separate functon `watchContract` or similar that will allow it.